app/services/attribute_uri_converter.rb

```diff
@@ -0,0 +1,47 @@
+class AttributeURIConverter
+  def initialize(attributes)
+    @attributes = attributes.dup
+  end
+
+  def convert_attributes
+    @attributes.each do |property, values|
+      convert_attribute(property, values)
+    end
+
+    @attributes
+  end
+
+  private
+
+  def convert_attribute(property, values)
+    case values
```

Comment:
Not a fan of the type checking - `values = convert_attribute_list(Array(values))`

Or maybe pass it in as an array always?